DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/FlowPartSyncer.cs

```diff
@@ -0,0 +1,257 @@
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Flows.Models;
+
+namespace DFC.ServiceTaxonomy.GraphSync.GraphSyncers.Parts
+{
+    //todo: there's an opportunity to share code with bagpart
+
+#pragma warning disable S1481 // need the variable for the new using syntax, see https://github.com/dotnet/csharplang/issues/2235
+
+    public class FlowPartGraphSyncer : IContentPartGraphSyncer
+    {
+        private readonly IContentFieldsGraphSyncer _contentFieldsGraphSyncer;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly Dictionary<string, ContentTypeDefinition> _contentTypes;
+
+        public string PartName => nameof(FlowPart);
+
+        private const string Widgets = "Widgets";
+        private const string FlowMetaData = "FlowMetadata";
+        private const string Alignment = "Alignment";
+        private const string Size = "Size";
+        private const string Ordinal = "Ordinal";
+
+        private static Func<string, string> _flowFieldsPropertyNameTransform = n => $"flow_{n}";
+
+        public FlowPartGraphSyncer(
+            IContentDefinitionManager contentDefinitionManager,
+            IContentFieldsGraphSyncer contentFieldsGraphSyncer,
+            IServiceProvider serviceProvider)
+        {
+            _contentFieldsGraphSyncer = contentFieldsGraphSyncer;
+            _serviceProvider = serviceProvider;
+
+            _contentTypes = contentDefinitionManager
+                .ListTypeDefinitions()
+                .Where(x => x.Parts.Any(p => p.Name == nameof(GraphSyncPart)))
+                .ToDictionary(x => x.Name);
+        }
+
+        public async Task AddSyncComponents(
+            dynamic content,
+            IMergeNodeCommand mergeNodeCommand,
+            IReplaceRelationshipsCommand replaceRelationshipsCommand,
+            ContentTypePartDefinition contentTypePartDefinition,
+            IGraphSyncHelper graphSyncHelper)
+        {
+            await AddWidgetsSyncComponents(content, replaceRelationshipsCommand, graphSyncHelper);
+
+            // FlowPart allows part definition level fields, but values are on each FlowPart instance
+            // prefix flow field property names, so there's no possibility of a clash with the eponymous fields property names
+            using var _ = graphSyncHelper.PushPropertyNameTransform(_flowFieldsPropertyNameTransform);
+
+            await _contentFieldsGraphSyncer.AddSyncComponents(
+                content,
+                mergeNodeCommand,
+                replaceRelationshipsCommand,
+                contentTypePartDefinition,
+                graphSyncHelper);
+        }
+
+        private async Task AddWidgetsSyncComponents(
+            dynamic content,
+            IReplaceRelationshipsCommand replaceRelationshipsCommand,
+            IGraphSyncHelper graphSyncHelper)
+        {
+            int widgetOrdinal = 0;
+            foreach (JObject? contentItem in content[Widgets])
+            {
```

Comment:
Should never happen but you've got a nullable JObject coming into the foreach and then telling the compiler it will never be null with contentItem! might just want to add a null check at the top.

Reply:
While there's nothing stopping an IEnumerator's Current method from returning null, in practice, using standard enumerators and foreach, I think it's pretty safe to ignore the possibility of null.